network/src/peer_store/peer_store_db.rs

```diff
@@ -0,0 +1,95 @@
+use crate::{
+    errors::{Error, PeerStoreError},
+    peer_store::{
+        addr_manager::AddrManager,
+        ban_list::BanList,
+        types::{AddrInfo, BannedAddr},
+        PeerStore,
+    },
+};
+use ckb_logger::debug;
+use std::fs::{create_dir_all, File, OpenOptions};
+use std::path::Path;
+
+const DEFAULT_ADDR_MANAGER_DB: &str = "addr_manager.db";
+const DEFAULT_BAN_LIST_DB: &str = "ban_list.db";
+
+impl AddrManager {
+    pub fn load<P: AsRef<Path>>(path: P) -> Result<Self, Error> {
+        let file = File::open(path)?;
+        let addrs: Vec<AddrInfo> = serde_json::from_reader(file).map_err(PeerStoreError::Serde)?;
+        let mut addr_manager = AddrManager::default();
+        addrs.into_iter().for_each(|addr| addr_manager.add(addr));
+        Ok(addr_manager)
+    }
+
+    pub fn dump<P: AsRef<Path>>(&self, path: P) -> Result<(), Error> {
+        let addrs: Vec<_> = self.addrs_iter().collect();
+        if let Some(dir) = path.as_ref().parent() {
+            create_dir_all(dir)?;
+        }
+        let file = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .append(true)
```

Comment:
it will create duplicate address
```suggestion
            .append(false)
```